src/svtk/svtk/utils/genotype_merging.py

```diff
@@ -23,41 +23,31 @@ def choose_best_genotype(sample, records):
     best_record : pysam.VariantRecord
     """
 
-    best_GT = (0, 0)
-    best_GQ = 0
-    best_record = None
-
-    if not any(['GQ' in record.samples[sample].keys() for record in records]):
-        for record in records:
-            if record.samples[sample]['GT'] != (0, 0):
-                return record
-        return records[0]
-
-    # Pick best non-reference genotype
-    # for record in records:
-    #    if record.samples[sample]['GQ'] >= best_GQ:
-        # if record is non-reference , use it
-        # or if it's a higher GQ for a reference call, use it
-    #        if record.samples[sample]['GT'] != (0, 0) or best_GT == (0, 0):
-    #            best_GT = record.samples[sample]['GT']
-    #            best_GQ = record.samples[sample]['GQ']
-    #            best_record = record
-
-    for record in records:
-        # if found non-ref GT, replace GT and GQ together
-        if record.samples[sample]['GT'] != (0, 0) and best_GT == (0, 0):
-            best_GT = record.samples[sample]['GT']
-            best_GQ = record.samples[sample]['GQ']
-            best_record = record
-        elif record.samples[sample]['GT'] == (0, 0) and best_GT != (0, 0):
-            continue
-        # if new GT  = best_GT, while found a higher GQ, replace GQ
+    # Returns true if the GT is called and non-ref
+    # TODO: does not handle multi-allelic CNVs
+    def _is_non_ref(format_fields):
+        return 'GT' in format_fields and sum(g for g in format_fields['GT'] if g is not None) > 0
+
+    def _get_mingq_index(format_fields_list):
+        gqs = [f['GQ'] for f in format_fields_list]
+        min_gq = min([x for x in gqs if x is not None], default=None)
```

Comment:
No need to construct the list, min can operate on generator
```suggestion
        min_gq = min((x for x in gqs if x is not None), default=None)
```

Reply:
Done